idaes/surrogate/plotting/sm_plotter.py

```diff
@@ -182,55 +195,62 @@ def parity(zdata, zfit, zlabels=None, clo=None, chi=None,
+        else:
+            name = filename
+        pdf = PdfPages(name)
+
     for j in range(numouts):  # loop over all outputs, zj
-        fig.append(plt.figure())
-        ax.append(fig[j].add_subplot())
-
-        # ax[j].plot(zdata[j, :], zdata[j, :], c='grey', label='Data')
-        # ax[j].scatter(zdata[j, :], zfit[j, :], s=3, c='b', marker='.',
-        #               label='Predictions')
-        ax[j].plot(zdata[:, j], zdata[:, j], c='grey', label='Data')
-        ax[j].scatter(zdata[:, j], zfit[:, j], marker='.', label='Predictions')
-
-        # plot confidence intervals if given data
-        if clo is not None and chi is not None:
-            ax[j].scatter(zdata[:, j], clo[:, j], c='g',
-                          marker='.', label=clabel)
-            ax[j].scatter(zdata[:, j], chi[:, j], c='g',
-                          marker='.', label=clabel)
-
-        ax[j].set_xlabel(zlabels[j])
-        ax[j].set_ylabel('Model Output')
-        ax[j].set_title('Parity Plot')
-        ax[j].legend()
+        fig = plt.figure()
+        ax = fig.add_subplot()
+
+        ax.plot(zdata[:, j], zdata[:, j], c='grey', label='Data')
+        ax.scatter(zdata[:, j], zfit[:, j], s=3, c='b', marker='.',
+                   label='Predictions')
+
+        ax.set_xlabel(zlabels[j])
+        ax.set_ylabel('Model Output')
+        ax.set_title('Parity Plot')
+        ax.legend()
 
         if show is True:
             plt.show()
-    if PDF is True:
-        pdfPrint(fig, filename)
+        if filename is not None:
+            pdf.savefig(fig)
+    pdf.close()
+
 
-def surrogate_residual(surrogate, dataframe, filename=None, relative_error=False):
+def surrogate_residual(surrogate, dataframe, filename=None,
+                       relative_error=False):
     input_data = dataframe[surrogate.input_labels()]
     output_data = dataframe[surrogate.output_labels()]
-    output_surrogate = surrogate.evaluate_surrogate(dataframe)
+    output_surrogate = surrogate.evaluate_surrogate(input_data)
     error = np.abs(output_data - output_surrogate)
-    if relative_error == True:
-        error = np.divide(error, np.maximum(output_data,1.0))
-    if filename is None:
-        residual(xdata=input_data.values, e=error.values, xlabels=surrogate.input_labels(), elabels=surrogate.output_labels())
-    else:
-        residual(xdata=input_data.values, e=error.values, xlabels=surrogate.input_labels(), elabels=surrogate.output_labels(),
-                 PDF=True, filename=filename)
+    if relative_error is True:
+        error = np.divide(error, np.maximum(output_data, 1.0))
+    _residual(xdata=input_data.values, error=error.values,
+              xlabels=surrogate.input_labels(),
+              elabels=surrogate.output_labels(),
+              filename=filename)
 
 
-def residual(xdata, e, xlabels=None, elabels=None, show=True, PDF=False,
-             filename='results_residual.pdf'):
+def _residual(xdata, error, xlabels=None, elabels=None, show=True,
```

Comment:
Instead of "error", would "residual" be a better name? When I see "error", I tend to think exception first.

Reply:
I agree, as long as it's not an issue with the argument sharing a name with the method.